feeder.go

```diff
@@ -112,71 +123,74 @@ func (f *Feeder) Import(path string) (FeederLoadResponse, error) {
 
 // Computes the RPMs images that have to be loaded into Docker
 // Returns a map with the repotag string as key and the name of the file as value
-func (f *Feeder) imagesToImport(path string) (map[string]string, error) {
-	rpmImages, err := findRPMImages(path)
+func (f *Feeder) imagesToImport(path string) (map[string]string, map[string][]string, error) {
+	rpmImages, rpmImageTags, err := findRPMImages(path)
 	if err != nil {
-		return rpmImages, err
+		return rpmImages, rpmImageTags, err
 	}
 
 	dockerImages, err := existingImages(f.dockerClient)
 	if err != nil {
-		return rpmImages, err
+		return rpmImages, rpmImageTags, err
 	}
 
 	for _, dockerImage := range dockerImages {
 		// ignore the tags that are already known by docker
 		delete(rpmImages, dockerImage)
+		delete(rpmImageTags, dockerImage)
 	}
 
-	return rpmImages, nil
+	return rpmImages, rpmImageTags, nil
 }
 
 // Finds all the Docker images shipped by RPMs
 // Returns a map with the repotag string as key and the full path to the
 // file as value.
-func findRPMImages(path string) (map[string]string, error) {
+func findRPMImages(path string) (map[string]string, map[string][]string, error) {
 	log.Debug("Finding images from %s", path)
 	walker := NewWalker(path, ".metadata")
 	images := make(map[string]string)
+	image_tags := make(map[string][]string)
 
 	if err := filepath.Walk(path, walker.Scan); err != nil {
-		return images, err
+		return images, image_tags, err
 	}
 
 	for _, file := range walker.Files {
 		file_path := filepath.Join(path, file)
-		repotag, image, err := repotagFromRPMFile(file_path)
+		repotag, repotags, image, err := repotagFromRPMFile(file_path)
 		if err != nil {
-			return images, err
+			return images, image_tags, err
 		}
 		// Check if image exist on disk
 		image_path := filepath.Join(path, image)
 		if _, err := os.Stat(image_path); err == nil {
 			images[repotag] = image_path
+			image_tags[repotag] = repotags
 		} else {
 			log.Debug("Image %s does not exist", image_path)
 		}
 	}
 
-	return images, nil
+	return images, image_tags, nil
 }
 
 // Compute the repotag (`<name>:<tag>`) starting from the name of the tar.xz
 // file shipped by RPM
 // Returns repotag (`<name>:<tag>`) and image name
-func repotagFromRPMFile(file string) (string, string, error) {
+func repotagFromRPMFile(file string) (string, []string, string, error) {
 	data, err := ioutil.ReadFile(file)
 	if err != nil {
-		return "", "", err
+		return "", nil, "", err
 	}
 
 	var metadata MetadataType
 	if err := json.Unmarshal(data, &metadata); err != nil {
-		return "", "", err
+		return "", nil, "", err
 	}
 
 	repotag := metadata.Image.Name + ":" + metadata.Image.Tags[0]
 	image := metadata.Image.File
 
-	return repotag, image, nil
+	return repotag, metadata.Image.Tags[1:], image, nil
```

Comment:
Once again hardcoded value, better is to use regexp

Reply:
This one is because repotag contains the first tag in metadata.Image.Tags[0] and so we need to return the list of the other tags, excluded the first one.

Reply:
ok, cool